### PR TITLE
Juju 7123/add role type to internal openfga

### PR DESCRIPTION
**Pull request by ale8k**
## Description

Allows the removal of a role from OpenFGA and plugs them into resourceTypes, adds their relation constant, and tests we can relate users->role, user->group--group#member->assignee->role.

If I'm missing anything, please let me know, I haven't touched access since the introduction of cofga.

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [x] Covered by integration tests

### CODE REVIEW

File: internal/openfga/openfga.go [hunk 1]
```diff
@@ -308,6 +310,38 @@ func (o *OFGAClient) RemoveApplicationOffer(ctx context.Context, offer names.App
 	return nil
 }
 
+// RemoveRole removes a group.
```
Comment (kian99) [inline internal/openfga/openfga.go]: The docstrings and comments need to be updated.

Comment (kian99) [reply internal/openfga/openfga.go]: Also, how come there is no AddRole method?

Comment (ale8k) [reply internal/openfga/openfga.go]: We appear to be just using auth relation CLI wise, there's no explicit relation command? Unless I'm mistaken? In our remove though, we clean up the group. I wanted to keep these the same.

Comment (kian99) [reply internal/openfga/openfga.go]: Took another look, makes sense, this function is just cleaning up as you said. AddGroup in this file doesn't even make sense, you'd add a person to a group or a group to something, not just AddGroup.

**Review by kian99 — APPROVED** [review]

**Review by SimoneDutto — APPROVED** [review]
lgtm, as long as group docstring are fixed

File: internal/openfga/names/names.go [hunk 2]
```diff
@@ -102,9 +105,13 @@ func ConvertGenericTag(t names.Tag) *Tag {
 // applicationoffer resource, so we specify user as BlankKindTag("controller"))
 func BlankKindTag(kind string) (*Tag, error) {
 	switch kind {
-	case names.UserTagKind, jimmnames.GroupTagKind,
-		names.ControllerTagKind, names.ModelTagKind,
-		names.ApplicationOfferTagKind, names.CloudTagKind,
+	case names.UserTagKind,
+		jimmnames.GroupTagKind,
+		jimmnames.RoleTagKind,
```
Comment (alesstimec) [inline internal/openfga/names/names.go:110]: nitpick: sort alphabetically?

File: internal/openfga/names/names.go [hunk 3]
```diff
@@ -59,7 +61,8 @@ type ResourceTagger interface {
 		names.ModelTag |
 		names.ApplicationOfferTag |
 		names.CloudTag |
-		jimmnames.ServiceAccountTag
+		jimmnames.ServiceAccountTag |
```
Comment (alesstimec) [inline internal/openfga/names/names.go:64]: nitpick: sort alphabetically?

File: internal/openfga/openfga.go [hunk 4]
```diff
@@ -308,6 +310,38 @@ func (o *OFGAClient) RemoveApplicationOffer(ctx context.Context, offer names.App
 	return nil
 }
 
+// RemoveRole removes a role.
+func (o *OFGAClient) RemoveRole(ctx context.Context, role jimmnames.RoleTag) error {
+	// Remove all access to a group. I.e. user->group
```
Comment (alesstimec) [inline internal/openfga/openfga.go:315]: s/group/role/

File: internal/openfga/openfga.go [hunk 5]
```diff
@@ -308,6 +310,38 @@ func (o *OFGAClient) RemoveApplicationOffer(ctx context.Context, offer names.App
...
+	if err := o.removeTuples(
+		ctx,
+		Tuple{
+			Relation: ofganames.AssigneeRelation,
+			Target:   ofganames.ConvertTag(role),
+		},
+	); err != nil {
+		return errors.E(err)
+	}
+	// Next remove all access that a group had. I.e. group->model
```
Comment (alesstimec) [inline internal/openfga/openfga.go:325]: s/group/role/

**Review by alesstimec — CHANGES_REQUESTED** [review]

**Pull request merged**